hotel/static/angular-apps/hotel/app.js

```diff
@@ -0,0 +1,148 @@
+angular.module('hotel', ['ngRoute', 'magfest'])
+    .service('errorHandler', function () {
+        return function () {
+            alert('I AM ERROR');  // TODO: better error handling
+        };
+    })
+    .service('Hotel', function ($window) {
+        var self = {
+            lists: {
+                rooms: [],
+                assigned: [],
+                unassigned: [],
+                unconfirmed: [],
+                declined: [],
+                all_attendees: []
+            },
+            _set: function(dst, src) {
+                dst.splice.apply(dst, [0, dst.length].concat(src));
+            },
+            set: function(data) {
+                angular.forEach(self.lists, function(xs, name) {
+                    self._set(xs, data[name] || []);
+                });
+                angular.forEach(self.lists, function (xs, name) {
+                    if (name != 'rooms' && name != 'all_attendees') {
+                        self.lists.all_attendees.push.apply(self.lists.all_attendees, xs);
+                    }
+                });
+            },
+            get: function (name, id) {
+                for (var i=0, x; x=self.lists[name][i]; i++) {
+                    if (x.id === id) {
+                        return x;
+                    }
+                }
+            }
+        };
+        if ($window.ROOM_DUMP) {
+            self.set(ROOM_DUMP);
+        }
+        return self;
+    })
+    .config(function($routeProvider){
+        $routeProvider
+            .when('/', {controller: 'HotelController', templateUrl: '../static/angular-apps/hotel/schedule.html'})
+            .when('/create-room', {controller: 'CreateController', templateUrl: '../static/angular-apps/hotel/room_form.html'})
+            .when('/edit-room/:roomId', {controller: 'EditController', templateUrl: '../static/angular-apps/hotel/room_form.html'})
+            .when('/attendee/:id', {controller: 'AttendeeController', templateUrl: '../static/angular-apps/hotel/attendee.html'})
+            .otherwise({redirectTo: '/'});
+    })
+    .controller('HotelController', function($scope, $http, Hotel, errorHandler) {
+        $scope.wrongNights = function (room, attendee) {
+            return room.nights != attendee.nights;
+        };
+        $scope.remove = function (attendee_id) {
+            $http({
+                method: 'post',
+                url: 'unassign_from_room',
+                params: {attendee_id: attendee_id}
+            }).success(Hotel.set).error(errorHandler);
+        };
+        $scope.deleteRoom = function (room_id) {
+            $http({
+                method: 'post',
+                url: 'delete_room',
+                params: {id: room_id}
+            }).success(Hotel.set).error(errorHandler);
+        };
```

Comment:
This was the only new function added to this Angular app, since the concept of locking-in rooms was new.